instrumentation/aws-lambda-1.0/library/src/main/java/io/opentelemetry/instrumentation/awslambda/v1_0/WrappedLambda.java

```diff
@@ -92,24 +92,38 @@ Method getRequestTargetMethod() {
        - Context can be omitted
        - Select the method with the largest number of parameters.
        - If two or more methods have the same number of parameters, AWS Lambda selects the method that has the Context as the last parameter.
+       - Non-Bridge methods are preferred
        - If none or all of these methods have the Context parameter, then the behavior is undefined.
+
+       Examples:
+       - handleA(String, String, Integer), handleB(String, Context) - handleA is selected (number of parameters)
+       - handleA(String, String, Integer), handleB(String, String, Context) - handleB is selected (has Context as the last parameter)
+       - generic method handleG(T, U, Context), implementation (T, U - String) handleA(String, String, Context), bridge method handleB(Object, Object, Context) - handleA is selected (non-bridge)
     */
     List<Method> methods = Arrays.asList(targetClass.getMethods());
     Optional<Method> firstOptional =
         methods.stream()
             .filter((Method m) -> m.getName().equals(targetMethodName))
             .sorted(
                 (Method a, Method b) -> {
-                  // sort descending (reverse of default ascending)
+                  // larger number of params wins
                   if (a.getParameterCount() != b.getParameterCount()) {
                     return b.getParameterCount() - a.getParameterCount();
                   }
-                  if (isLastParameterContext(a.getParameters())) {
-                    return -1;
-                  } else if (isLastParameterContext(b.getParameters())) {
-                    return 1;
+                  boolean firstCtx = isLastParameterContext(a.getParameters());
+                  boolean secondCtx = isLastParameterContext(b.getParameters());
+                  // only one of the methods has last param context ?
+                  if (firstCtx ^ secondCtx) {
```

Comment:
personally I'd avoid using rarely used boolean operators like this. I had to look up that this was an xor. It makes the code harder to read.

Reply:
I also recommend putting all of this code into a method with a clear name. I honestly have no idea what the purpose of all this stuff is.

Reply:
Also, are there tests for these changes?

Reply:
Honestly I had the same reaction, including the xor double take. The added javadoc helps a lot but if this can be restructured for more readability that would be nice - if it was just me whatever, but seems to be more general :)

Reply:
Yeah, I'm not fond of XOR either, however in this context - mutual exclusion - seemed to fit the bill nicely. However can restructure as requested.